### PR TITLE
Add port tagging

### PR DESCRIPTION
**Pull request by kon-angelo**
**What this PR does / why we need it**:

Allow to tag neutron resources with the cluster tags.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```improvement user
Created Neutron ports will now be tagged with the shoot tags.
```

### CODE REVIEW

**Review by AxiomSamarth — APPROVED** [review]
/lgtm

File: pkg/driver/executor/executor.go [hunk 1]
```diff
@@ -492,22 +495,10 @@ func (ex *Executor) getMachineByID(_ context.Context, serverID string) (*servers
...
-		}
-	}
-
-	if searchClusterName == "" || searchNodeRole == "" {
-		klog.Warningf("getMachineByName operation can not proceed: cluster/role tags are missing for machine [Name=%q]", machineName)
-		return nil, fmt.Errorf("getMachineByName operation can not proceed: cluster/role tags are missing for machine [Name=%q]", machineName)
+	searchClusterName, searchNodeRole, ok := findMandatoryTags(ex.Config.Spec.Tags)
+	if !ok {
+		klog.Warningf("operation can not proceed: cluster/role tags are missing")
+		return nil, fmt.Errorf("operation can not proceed: cluster/role tags are missing")
```
Comment (himanshu-kun) [inline pkg/driver/executor/executor.go:501]: we can enhance the error message to including `machineName` in the error message like it was before

Comment (himanshu-kun) [reply pkg/driver/executor/executor.go:501]: and also in the warning message

**Review by himanshu-kun — CHANGES_REQUESTED** [review]
just minor changes otherwise looks perfect

**Review by himanshu-kun — APPROVED** [review]
/lgtm

**Pull request merged**